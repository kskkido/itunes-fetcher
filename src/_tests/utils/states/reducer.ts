import { Action } from 'redux';
import { rootReducer, RootState } from 'states/root/reducer';
import { noop } from './action';

// TODO fix types
export const batchedActionsReducer = (
  state: RootState | {},
  actions: Action[] = [noop]
) => (
  actions.reduce(
    rootReducer,
    state as RootState
  ) as RootState
);
