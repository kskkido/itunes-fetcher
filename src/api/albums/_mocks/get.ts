import { GetPayload, Album } from '../types';

const createAlbum = (id: number): Album => ({
  wrapperType: '',
  artistId: id,
  collectionId: id,
  artistName: '',
  collectionName: '',
  collectionCensoredName: '',
  artistViewUrl: '',
  collectionViewUrl: '',
  artworkUrl60: '',
  artworkUrl100: '',
  collectionPrice: '',
  collectionExplicitness: '',
  discCount: 0,
  discNumber: 0,
  trackCount: 0,
  country: '',
  currency: '',
  primaryGenreName: '',
});

export const payload: GetPayload = {
  resultCount: 5,
  results: [0,1,2,3,4].map(createAlbum),
};
