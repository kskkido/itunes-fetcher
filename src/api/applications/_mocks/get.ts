import { GetPayload, Application } from '../types';

const createApplication = (id: number): Application => ({
  wrapperType: '',
  artistId: 0,
  artistName: '',
  artworkUrl60: '',
  artworkUrl100: '',
  country: '',
  currency: '',
  primaryGenreName: '',
  trackId: id,
  trackName: '',
  trackViewUrl: '',
});

export const payload: GetPayload = {
  resultCount: 5,
  results: [0,1,2,3,4].map(createApplication),
};
