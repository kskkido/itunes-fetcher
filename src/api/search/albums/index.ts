import apiFetch from 'api/utils/apiFetch';
import { ITUNES_SEARCH_API } from 'constants/urls';
import { SearchParams, SearchPayload } from './types';

export default {
  get: (
    params: SearchParams
  ): Promise<
    SearchPayload
  > => (
    apiFetch.get({
      params: {
        ...params,
        entity: 'album',
        country: 'US',
      },
      url: ITUNES_SEARCH_API,
    })
  ),
};
