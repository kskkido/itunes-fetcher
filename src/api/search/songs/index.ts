import apiFetch from 'api/utils/apiFetch';
import { SearchParams, SearchPayload } from './types';

export default {
  get: (
    params: SearchParams
  ): Promise<
    SearchPayload
  > => (
    apiFetch.get({
      params: {
        ...params,
        entity: 'song',
        country: 'US',
      },
      url: 'https://itunes.apple.com/search',
    })
  ),
};
