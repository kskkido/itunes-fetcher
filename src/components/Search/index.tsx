import React from 'react';
import AsyncSearchContainer from 'containers/Search/Parts/AsyncSearchContainer';
import SearchForm from 'containers/Search/Parts/Forms/SearchForm';
import Albums from 'containers/Search/Parts/Results/Albums';
import Songs from 'containers/Search/Parts/Results/Songs';
import PageContainer from 'components/Shared/Layouts/PageContainer';
import Section from 'components/Shared/Section';

const Search: React.SFC = () => (
  <PageContainer>
    <Section>
      <SearchForm />
    </Section>
    <AsyncSearchContainer>
      <Songs />
      <Albums />
    </AsyncSearchContainer>
  </PageContainer>
);

export default Search;
