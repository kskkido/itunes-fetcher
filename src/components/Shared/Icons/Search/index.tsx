import React from 'react';
import Icon from 'react-native-vector-icons/Feather';

type SearchProps = {
  size: number;
};

const Search: React.SFC<SearchProps> = ({
  size,
}) => (
  <Icon
    name="search"
    size={size}
  />
);

export default Search;
