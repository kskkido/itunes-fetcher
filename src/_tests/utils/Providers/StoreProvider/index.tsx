import React from 'react';
import { Provider } from 'react-redux';
import { RootState } from 'states/root/reducer';
import { configureStore } from 'states';

type StoreProviderProps = {
  children: React.ReactNode;
  state: Partial<RootState>;
};

const StoreProvider: React.SFC<StoreProviderProps> = ({
  children,
  state,
}) => (
  <Provider store={configureStore(state)}>
    {children}
  </Provider>
);

export default StoreProvider;
