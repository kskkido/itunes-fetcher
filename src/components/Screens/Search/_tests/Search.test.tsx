/// <reference types="jest" />
import React from 'react';
import TestRenderer from 'react-test-renderer';
import StoreProvider from '_tests/utils/Providers/StoreProvider';
import * as albumsGetMock from 'api/albums/_mocks/get';
import * as applicationsGetMock from 'api/applications/_mocks/get';
import * as songsGetMock from 'api/songs/_mocks/get';
import * as albumsActions from 'states/albums/actions';
import * as applicationsActions from 'states/applications/actions';
import * as songsActions from 'states/songs/actions';
import { batchedActionsReducer } from '_tests/utils/states/reducer';
import Search from '../';

const setup = () => ({
  tree: TestRenderer.create(
    <StoreProvider state={
      batchedActionsReducer(
        {},
        [
          albumsActions.albumsReceive(albumsGetMock.payload),
          applicationsActions.applicationsReceive(applicationsGetMock.payload),
          songsActions.songsReceive(songsGetMock.payload),
        ]
      )}>
      <Search />
    </StoreProvider>
  ),
});

describe('Search', () => {
  it('should render as expected', () => {
    expect(setup().tree.toJSON()).toMatchSnapshot();
  });
});
