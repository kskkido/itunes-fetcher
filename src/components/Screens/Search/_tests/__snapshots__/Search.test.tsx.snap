// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render as expected 1`] = `
<RCTScrollView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "flexDirection": "column",
          "paddingHorizontal": 12,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#eee",
              "borderRadius": 4,
              "flexDirection": "row",
              "height": 45,
              "paddingLeft": 10,
              "width": "100%",
            }
          }
        >
          <Icon
            name="search"
            size={20}
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <TextInput
            allowFontScaling={true}
            numberOfLines={1}
            onChangeText={[Function]}
            onEndEditing={[Function]}
            placeholder="Search anything"
            rejectResponderTermination={true}
            style={
              Object {
                "width": "100%",
              }
            }
            underlineColorAndroid="transparent"
          />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "flexDirection": "column",
          "paddingHorizontal": 12,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
              "fontWeight": "600",
              "marginBottom": 0,
              "textAlign": "left",
            }
          }
        >
          Songs
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontWeight": "700",
            }
          }
        >
          See All...
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
            "marginTop": 5,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "flexDirection": "column",
          "paddingHorizontal": 12,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
              "fontWeight": "600",
              "marginBottom": 0,
              "textAlign": "left",
            }
          }
        >
          Albums
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontWeight": "700",
            }
          }
        >
          See All...
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
            "marginTop": 5,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#ddd",
          "borderBottomWidth": 1,
          "flexDirection": "column",
          "paddingHorizontal": 12,
          "paddingVertical": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 40,
              "fontWeight": "600",
              "marginBottom": 0,
              "textAlign": "left",
            }
          }
        >
          Apps
        </Text>
        <Text
          onPress={[Function]}
          style={
            Object {
              "fontWeight": "700",
            }
          }
        >
          See All...
        </Text>
      </View>
      <View
        style={
          Object {
            "marginBottom": 5,
            "marginTop": 5,
            "width": "100%",
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "paddingHorizontal": 12,
            "paddingVertical": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "marginBottom": 5,
              "marginTop": 5,
              "width": "100%",
            }
          }
        />
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "height": 60,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "",
              }
            }
            style={
              Object {
                "backgroundColor": "#ddd",
                "height": "100%",
                "width": 60,
              }
            }
          />
          <View
            style={
              Object {
                "height": "100%",
                "marginLeft": 5,
                "marginRight": 5,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "center",
              }
            }
          >
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 35,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
            <Text
              numberOfLines={1}
              style={
                Object {
                  "flex": 1,
                  "fontSize": 28,
                  "fontWeight": "600",
                  "marginBottom": 0,
                  "textAlign": "left",
                }
              }
            >
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
